vulnerability/worker.go

```diff
@@ -0,0 +1,142 @@
+	"github.com/elastic/cloudbeat/config"
+	"github.com/elastic/cloudbeat/resources/providers/awslib/ec2"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type VulnerabilityWorker struct {
+	log      *logp.Logger
+	cfg      *config.Config
+	provider *ec2.Provider
+	// cloudIdentity *awslib.Identity
+	fetcher    VulnerabilityFetcher
+	replicator VulnerabilityReplicator
+	verifier   VulnerabilityVerifier
+	evaluator  VulnerabilityEvaluator
+	wg         sync.WaitGroup
+}
+
+func NewVulnerabilityWorker(log *logp.Logger, c *config.Config) (*VulnerabilityWorker, error) {
+	log.Debug("VulnerabilityWorker: New")
+	// ctx := context.Background()
+
+	// acp := awslib.ConfigProvider{MetadataProvider: awslib.Ec2MetadataProvider{}}
+	// awsConfig, err := acp.InitializeAWSConfig(ctx, c.CloudConfig.AwsCred)
+	// log.Info("VulnerabilityWorker: New - InitializeAWSConfig")
+	awsConfig, err := libbeat_aws.InitializeAWSConfig(c.CloudConfig.AwsCred)
+	if err != nil {
+		return nil, fmt.Errorf("VulnerabilityWorker: failed to initialize AWS credentials: %w", err)
+	}
+
+	// identityProvider := awslib.GetIdentityClient(awsConfig)
+	// identity, err := identityProvider.GetIdentity(ctx)
+	// if err != nil {
+	// 	return nil, fmt.Errorf("VulnerabilityWorker: could not get cloud indentity: %w", err)
+	// }
+
+	// provider := ec2.NewEC2Provider(log, *identity.Account, awsConfig)
+	// log.Info("VulnerabilityWorker: New - NewEC2Provider")
+	provider := ec2.NewEC2Provider(log, "", awsConfig)
+
+	// log.Info("VulnerabilityWorker: New - NewVulnerabilityFetcher")
+	fetcher := NewVulnerabilityFetcher(log, provider)
+	// log.Info("VulnerabilityWorker: New - NewVulnerabilityReplicator")
+	replicator := NewVulnerabilityReplicator(log, provider)
+	// log.Info("VulnerabilityWorker: New - NewVulnerabilityVerifier")
+	verifier := NewVulnerabilityVerifier(log, provider)
+	// log.Info("VulnerabilityWorker: New - NewVulnerabilityEvaluator")
+	evaluator, err := NewVulnerabilityEvaluator(log)
+	if err != nil {
+		return nil, fmt.Errorf("VulnerabilityWorker: could not get init NewVulnerabilityEvaluator: %w", err)
+	}
+
+	// log.Info("VulnerabilityWorker: New - NewVulnerabilityWorker")
+	return &VulnerabilityWorker{
+		log:      log,
+		cfg:      c,
+		provider: provider,
+		// cloudIdentity: identity,
+		fetcher:    fetcher,
+		replicator: replicator,
+		verifier:   verifier,
+		evaluator:  evaluator,
+		wg:         sync.WaitGroup{},
+	}, nil
+}
+
+// // TODO: Maybe verify more than one snapshot
+// func (f *VulnerabilityWorker) Run(ctx context.Context) error {
+// 	f.log.Info("Starting VulnerabilityWorker.Run")
+// 	go f.work(ctx)
+// 	return nil
+// }
+
+func (f *VulnerabilityWorker) Run(ctx context.Context) {
+	// TODO: Handle deletion of snapshots
+	f.log.Info("Starting VulnerabilityWorker.work")
+	for {
+		select {
+		case <-ctx.Done():
+			f.log.Info("VulnerabilityWorker.work context canceled")
+			return
+		default:
+			f.wg.Add(1)
+			go func() {
```

Comment:
We can also use the `pipeline` package to implement that flow to align with other parts in cloudbeat.
We can extend it to have more functionality or in a different PR we can replace it with another RX package for go.